addons/purchase_stock/models/purchase_order_line.py

```diff
@@ -312,6 +312,8 @@ def _prepare_account_move_line(self, move=False):
                 self.company_id.currency_id,
                 round=False,
             )
+        if 'amount_currency' not in res:
+            res['amount_currency'] = self.price_unit_discounted * (self.qty_received or 1)
```

Comment:
If we go for that solution, don't you think it would make more sense to use `qty_to_invoice`, as done in `super` ?

Reply:
It is to stay consistent with `balance` that is set just above as the 2 values should be linked.